File: Python/6py 1/bingo.py
"""
|--------------------------------------------------------------------------
| Lottery Ticket
|--------------------------------------------------------------------------
|
"""

def bingo(ticket, win):
   res = sum(1 for string, num in ticket if any(ord(char) == num for char in string))
   return "Winner!" if res >= win else "Loser!"


def bingo(ticket, win):
   return 'Winner!' if sum(chr(n) in s for s, n in ticket) >= win else 'Loser!'


def bingo(ticket,win):
   counter = 0
   for x,y in ticket:
      for i in x: counter += 1 if ord(i) == y else 0 
   return 'Loser!' if counter < win else 'Winner!'


print(bingo([['ABC', 65], ['HGR', 74], ['BYHT', 74]], 2), 'Loser!')
print(bingo([['ABC', 65], ['HGR', 74], ['BYHT', 74]], 1), 'Winner!')
print(bingo([['HGTYRE', 74], ['BE', 66], ['JKTY', 74]], 3), 'Loser!')

File: PHP/6php 1/bingo.php
<?php
/*
|--------------------------------------------------------------------------
| Lottery Ticket
|--------------------------------------------------------------------------
|
*/

function bingo($ticket, $win) {
   $res = array_reduce($ticket, function($count, $item) {
      [$string, $num] = $item;
      return $count + (preg_match('/' . chr($num) . '/', $string) ? 1 : 0);
   }, 0);
   return $res >= $win ? "Winner!" : "Loser!";
}


function bingo(array $ticket, int $win): string {
   return count(array_filter( $ticket, function ($arr) { 
      return substr_count($arr[0], chr($arr[1])); 
   }))>=$win ? 'Winner!' : 'Loser!';
}


function bingo(array $ticket, int $win): string {
   $result = 0;
   foreach($ticket as $one) {
      $result += strpos($one[0], chr($one[1])) !== false ? 1 : 0;
   }
   return $result >= $win ? "Winner!" : "Loser!";
}


print_r(bingo([['ABC', 65], ['HGR', 74], ['BYHT', 74]], 2) . ' Loser!
');
print_r(bingo([['ABC', 65], ['HGR', 74], ['BYHT', 74]], 1) . ' Winner!
');
print_r(bingo([['HGTYRE', 74], ['BE', 66], ['JKTY', 74]], 3) . ' Loser!
');